src/lang/nodejs/src/index.test.js

```diff
@@ -0,0 +1,191 @@
+const blot = require('./index.js');
+const WebSocket = require('ws');
+const fs = require('fs');
+let ws = null;
+
+beforeAll(() => {
+  blot.setPort(3000).openManually();
+  ws = new WebSocket('ws://localhost:3000');
+});
+
+afterAll(() => {
+  ws.close();
+});
+
+describe('Non JSON', () => {
+
+  test('undefined', () => {
+    expect(() => blot.visualise(undefined)).toThrow('Visualise must take in a valid JSON value')
+  });
+
+  test('Function', () => {
+    expect(() => blot.visualise(() => "hello world")).toThrow('Visualise must take in a valid JSON value');
+  });
+});
+
+describe('Standard JSON tests', () => {
+
+  test('Tests empty string', (done) => {
+
+    ws.onmessage = msg => {
+      expect(msg.data).toBe('');
+      done();
+    };
+
+    blot.visualise('');
```

Comment:
Let's declare a variable for the data and use that in both places instead.

Reply:
ok sure. I'm curious what's the rationale behind this

Reply:
It is to avoid repeating it and make the data that is being tested a little more prominent. Let's do this for all the tests.